gpusim.cpp

```diff
@@ -212,17 +220,17 @@ bool GPUSimServer::setupSocket(const QString& socket_name)
 }
 
 void GPUSimServer::similaritySearch(const Fingerprint& reference,
-        const QString& dbname,
+        const QString& dbname, const QString& dbkey,
```

Comment:
Does it make sense to pass these things around as a pair?

Reply:
Eh, I'm hesitant.  Here it would involve either later having to add more variables to reference `.first` and `.second`, or just using the hard to read `.first` and `.second` later.  I think it would actually hurt readability.  If I could switch the dict to be a vector of pairs I'd be more tempted, but I need the lookup ability.

Reply:
Oh, I was just proposing that you pass the name_key_pair directly from searchDatabases() and resolve here:

const auto& dbname = dbname_key_pair.first;
const auto& dbkey = dbname_key_pair.second;
if(calc_type == CalcType::GPU) {
    ...